pulpcore/app/models/repository.py

```diff
@@ -93,6 +93,18 @@ def save(self, *args, **kwargs):
             if adding:
                 self.create_initial_version()
 
+                # lock the repository if it was created from within a running task
+                try:
+                    task_id = environ["PULP_TASK_ID"]
+                except KeyError:
+                    return
+
+                repository_url = Value(get_url(self))
+                update_func = Func(
+                    F("reserved_resources_record"), repository_url, function="ARRAY_APPEND"
+                )
```

Comment:
```suggestion
                repository_url = get_url(self)
                update_func = Func(
                    F("reserved_resources_record"), Value(repository_url), function="ARRAY_APPEND"
                )
```
Feel free to ignore. Just a feeling the `F` and `Value` are kind of a similar thing.